File: Frontend/src/types/memorial-standard.ts
export interface MemorialStandard {
  id: string;
  name: string;
  description: string;
  standardText: string;
  promptTemplate: string;
  active: boolean;
  isDefault: boolean;
  ownerId: string;
  ownerName: string;
  createdAt: string;
  updatedAt: string;
}

export interface MemorialStandardCreate {
  name: string;
  description: string;
  standardText: string;
  promptTemplate: string;
  isDefault: boolean;
}

export interface MemorialStandardFormData {
  name: string;
  description: string;
  standardText: string;
  promptTemplate: string;
  isDefault: boolean;
}

File: Backend/src/main/java/com/momorialPro/CadMemorial/model/MemorialStandard.java
package com.momorialPro.CadMemorial.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.UUID;

@Entity
@Table(name = "tab_memorial_standards")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class MemorialStandard extends AuditBase {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "standard_id", updatable = false, nullable = false)
    private UUID id;

    @Column(name = "name", nullable = false, length = 100)
    private String name;

    @Column(name = "description", length = 500)
    private String description;

    @Column(name = "standard_text", nullable = false, columnDefinition = "TEXT")
    private String standardText;

    @Column(name = "prompt_template", nullable = false, columnDefinition = "TEXT")
    private String promptTemplate;

    @Column(name = "active", nullable = false)
    @Builder.Default
    private Boolean active = true;

    @Column(name = "is_default", nullable = false)
    @Builder.Default
    private Boolean isDefault = false;

    // Relacionamento com o usuário que criou a norma
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "owner_id")
    private User owner;
}